tests/test_transformer.py

```diff
@@ -37,22 +43,45 @@ def min_time_limit(self, cqm):
         return 1
 
 
+class MockNL():
+    def sample(self, nl: Model, *, time_limit: float, label: str):
+        nl.states.resize(1)
+        results = np.array([])
+
+        for decision in nl.iter_decisions():
+            decision.set_state(0, FEASIBLE_NL_SOLUTION)
+            np.append(results, decision[0])
+
+        mock_results = np.asarray(results, dtype=bool)
+        return mock_results
```

Comment:
I don't think you ever use this return in the code, right? For simplification, I'd just return nothing or, perhaps more correct, return a `concurrent.futures.Future()` object (you should be able to just `import concurrent.futures` at the top.